### PR TITLE
[Thanos - S3 Browser] List S3 objects by prefix

### PR DESCRIPTION
**Pull request by grace-punzalan**
Testing in Thanos - Overriding s3 package locally. 

https://user-images.githubusercontent.com/98766127/191062484-aa487b8b-2bdd-4b71-a438-175cfe8724cc.mov

### CODE REVIEW

File: pkg/s3/s3_service.go [hunk 1]
```diff
@@ -86,10 +86,11 @@ func (ps3 *S3Client) ListBuckets(ctx context.Context) (*BucketLists, error) {
 	return &res, nil
 }
 
-func (ps3 *S3Client) ListFiles(ctx context.Context, bucket *string) (*FileLists, error) {
+func (ps3 *S3Client) ListFiles(ctx context.Context, bucket, prefix *string) (*FileLists, error) {
```
Comment (grace-punzalan) [inline pkg/s3/s3_service.go:89]: I don't think this is used anywhere at this point.
https://github.com/search?q=org%3Aphil-inc+ListFiles&type=code

Comment (Swechhya) [reply pkg/s3/s3_service.go:89]: @grace-punzalan  Maybe create a new function instead as prefix might not be required in some cases, or maybe check if nil prefix is supported?

Comment (grace-punzalan) [reply pkg/s3/s3_service.go:89]: I tested that an empty string prefix returns all the files in the bucket. Is that enough? If not I can create a new function too.

Comment (Swechhya) [reply pkg/s3/s3_service.go:89]: @grace-punzalan Then I think we can modify this function itself as it supports both purposes and isn't used anywhere at this point.

**Pull request merged**